### PR TITLE
feat: adding telemetry to the html report tab switching

### PR DESCRIPTION
**Pull request by olavtar**
Tab switching of the providers will be reported to Segment. 

Jira: https://issues.redhat.com/browse/APPENG-2329

### CODE REVIEW

File: ui/src/mocks/reportWithoutToken.mock.ts [hunk 1]
```diff
@@ -396,5 +396,5 @@ export const withoutTokenReport: AppData = {
       "    \"purl\": \"pkg:oci/ubi-minimal@sha256:06d06f15f7b641a78f2512c8817cbecaa1bf549488e273f5ac27ff1654ed33f0?repository_url=registry.access.redhat.com/ubi9/ubi-minimal&tag=9.3-1552&arch=amd64\",\n" +
       "    \"catalogUrl\": \"https://catalog.redhat.com/software/containers/ubi9/ubi-minimal/615bd9b4075b022acc111bf5?architecture=amd64&image=65a828e3cda4984705d45d26\"\n" +
       "  }\n" +
-      "]"
```
Comment (ruromero) [inline ui/src/mocks/reportWithoutToken.mock.ts:399]: It seems you're missing here the data similar to what you added to the mixedMock

File: ui/src/components/TabbedLayout.tsx [hunk 2]
```diff
@@ -1,14 +1,46 @@
...
+
+    useEffect(() => {
+      if (!analytics) return;
+
+      if (appContext.userId == null) {
+        if (appContext.anonymousId != null) {
+          analytics.setAnonymousId(appContext.anonymousId);
+        }
+      } else {
+        if (appContext.userId !== previousUserId.current) {
```
Comment (ruromero) [inline ui/src/components/TabbedLayout.tsx:27]: How can there be different `userId`s for the same report?

File: ui/src/components/TabbedLayout.tsx [hunk 3]
```diff
@@ -1,14 +1,46 @@
...
+          analytics.identify(appContext.userId);
```
Comment (ruromero) [inline ui/src/components/TabbedLayout.tsx:28]: The `identify` event should've been sent by the backend. You shouldn't need this. This event will override the existing user-related data.

File: ui/src/components/TabbedLayout.tsx [hunk 4]
```diff
@@ -1,14 +1,46 @@
...
+          previousUserId.current = appContext.userId;
+        }
+      }
+      const handleActiveTabKeyUpdate = async (newActiveTabKey: string | number) => {
+        if (newActiveTabKey !== previousActiveTabKey.current) {
+          analytics.track("rhda.exhort.tab", {
```
Comment (ruromero) [inline ui/src/components/TabbedLayout.tsx:34]: Can't you set the userId/anonymousId to the track event as we do in the backend?
Try using a `SegmentEvent` instead:
```
analytics.track({
  userId: appContext.userId,
  anonymousId: appContext.anonymousId,
  event: "rhda.exhort.tab"
}, {
  tabName: newActiveTabKey
});
```

I haven't tried it, it's just from what I saw in the JS api.

**Review by ruromero — COMMENTED** [review]
I've added a few comments with my questions

File: ui/src/mocks/reportWithoutToken.mock.ts (hunk 1 above)
Comment (olavtar) [reply ui/src/mocks/reportWithoutToken.mock.ts:399]: Those 3 fields are optional, so it does not have to be there in all the reports.

File: ui/src/components/TabbedLayout.tsx (hunk 3 above)
Comment (olavtar) [reply ui/src/components/TabbedLayout.tsx:28]: I'll remove it.

File: ui/src/mocks/reportWithoutToken.mock.ts (hunk 1 above)
Comment (ruromero) [reply ui/src/mocks/reportWithoutToken.mock.ts:399]: ok, the comma made me think you wanted to add it.

File: ui/src/components/TabbedLayout.tsx (hunk 2 above)
Comment (olavtar) [reply ui/src/components/TabbedLayout.tsx]: removed.

File: ui/src/components/TabbedLayout.tsx (hunk 3 above)
Comment (olavtar) [reply ui/src/components/TabbedLayout.tsx]: Identify() does not override

File: ui/src/components/TabbedLayout.tsx (hunk 4 above)
Comment (olavtar) [reply ui/src/components/TabbedLayout.tsx]: It does not change the value of it if I add it, only adds it as part of the property. Identify() changes the userId and setAnonymousId() changes anonymousId.

**Review by ruromero — APPROVED** [review]
LGTM

**Pull request merged**